### ISSUE TITLE
Fix FP S4123 (`no-invalid-await`): do not raise issue for `unknown` type

### ISSUE THREAD

**Issue by alexander-kamushkin-sonarsource** [SonarSource/SonarJS#3966]
Our current implementation does not cover `unknown` type, for which the issue should not be raised (same as `any`).
Reported in [community post](https://community.sonarsource.com/t/redundant-await-on-a-non-promise-can-give-a-false-positive/91594)

```ts
export class NoErrorThrownError extends Error {}

export class TestUtils {
    /**
     * A method to be able to get round eslint-jest
     * Avoid calling `expect` conditionally`  jest/no-conditional-expect
     *
     * @param call the function to call that an error is expected to be thrown.
     * @returns The error that was thrown or NoErrorThrownError if no error was thrown.
     */
    public static getError = async <TError>(
        call: () => PromiseLike<unknown> | unknown
    ): Promise<TError> => {
        try {
            // Do NOT remove the await here, it is needed as the call maybe awaitable.
            // The tests for this function proves it.
            await call();
            throw new NoErrorThrownError();
        } catch (error) {
            return error as TError;
        }
    };
}
`
